cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/service/http/BasicHttpServiceContext.java

```diff
@@ -123,6 +146,44 @@ public int getInstanceId() {
     return instanceId;
   }
 
+  @Override
+  public DynamicPluginConfigurer createPluginConfigurer() {
+    File tmpDir = new File(cConf.get(Constants.CFG_LOCAL_DATA_DIR),
+                           cConf.get(Constants.AppFabric.TEMP_DIR)).getAbsoluteFile();
+    try {
+      File pluginsDir = Files.createTempDirectory(tmpDir.toPath(), "plugins").toFile();
+      PluginInstantiator instantiator = new PluginInstantiator(cConf, program.getClassLoader(), pluginsDir);
+      PluginConfigurer configurer =
+        new DefaultPluginConfigurer(artifactId, namespaceId, instantiator, pluginFinder);
+      return new DynamicPluginConfigurer() {
+        @Nullable
+        @Override
+        public <T> T usePlugin(String pluginType, String pluginName, String pluginId, PluginProperties properties,
+                               PluginSelector selector) {
+          return configurer.usePlugin(pluginType, pluginName, pluginId, properties, selector);
+        }
+
+        @Nullable
+        @Override
+        public <T> Class<T> usePluginClass(String pluginType, String pluginName, String pluginId,
+                                           PluginProperties properties, PluginSelector selector) {
+          return configurer.usePluginClass(pluginType, pluginName, pluginId, properties, selector);
+        }
+
+        @Override
+        public void close() throws IOException {
+          try {
+            instantiator.close();
```

Comment:
also, it is better to hook this with the `BasicHttpServiceContext.close()` method (overriding from the parent one) so that if the user forget to close the configurer, it will eventually get closed (the context.close() will be called when a handler instance is no longer in use).